include/seqan3/io/sequence_file/input.hpp

```diff
@@ -691,6 +692,19 @@ class sequence_file_input
     }
     //!\}
 
+    //!\brief Returns a list of valid file extensions.
+    //!\returns std::vector over std::string with all valid file extensions specified by `valid_formats`.
+    static std::vector<std::string> valid_file_extensions()
+    {
+        std::vector<std::string> extensions;
```

Comment:
we will need this code in each of the files (6 in total currently) so to reduce code it would be great if this can be outsourced into a function in `io/detail/misc` :pray:

Reply:
I already proposed a small mixin class that adds this feature. But @h-2 was proposing to keep it simple. I am in general in favour for DRYing this out and don't see any points against it.

Reply:
Well, I think we should sit down and think about what else we can DRY about the files and do it in one step altogether; but if you really want to factor it out now, I would prefer not to do a CRTP-mixin, but rather just a free function in misc that is called by the member function.
CRTP-mixins have the drawback that they are difficult to document and I would rather hide the implementation details from the user, the class is complicated enough.

Reply:
Why a CRTP-mixin? I was aiming all along for a free function that takes the vallid_formats and returns a list of extensions.

Reply:
> I would prefer not to do a CRTP-mixin, but rather just a free function in misc that is called by the member function.

The interface still needs to be repeated in each file and still there can go a lot of  things wrong (e.g. diverging documentation and probably other subtle errors). And even if the interface is very simple not much can go wrong it suggests a wrong example that could be copied in other places for more complex problems. 

> CRTP-mixins have the drawback that they are difficult to document and I would rather hide the implementation details from the user, the class is complicated enough.

I think we don't need a full CRTP-mixin. Something that just get's the valid_formats type would be sufficient. I think this can be cleanly documented and it does not need to be detail stuff. Regarding further refactorings. We should discuss it, but given the huge amount of tasks and the few weeks we have left, I wouldn't go for something big now.

Reply:
I am also not in favour of discussing this too long, I would propose the following free functions in misc
```cpp
template <typename ... format_ts>
std::vector<std::string> valid_file_format_extensions()
{
    // impl
}

inline std::vector<std::string> valid_file_compression_extensions()
{
   // impl based on macros
#ifdef SEQAN3_HAS_GRZIP 
}
```

And then just use the free functions in the argParse and display something like
```
--input   A file with the input sequences. Valid format extensions are [fasta, fa, fastq, fq], possibly combined with [gz, bz2, bgzf].
```